tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/EndMarkerSuite.scala

```diff
@@ -0,0 +1,79 @@
+package scala.meta.tests.parsers.dotty
+
+import scala.meta.tests.parsers._
+import scala.meta._
+
+class EndMarkerSuite extends BaseDottySuite {
+
+  implicit val parseTempl: String => Stat = code => templStat(code)(dialects.Dotty)
+
+  val parseSource: String => Source = code => source(code)(dialects.Dotty)
+
+  test("end-marker") {
+    runTestAssert[Stat]("end token")(
+      Term.EndMarker(Term.Name("token"))
+    )
+  }
+
+  test("end-marker-keyword") {
+    val markers =
+      List("if", "while", "for", "match", "try", "new", "this", "given", "extension", "val")
+    for (m <- markers) {
+      parseTempl(s"end ${m}")
+    }
+  }
+
+  test("end-marker-toplevel") {
+    //NOTE(kpbochenek): Needs to be changed to object a: ... end a
+    // after significant indentation is present
+    val code = """|object a { }
+                  |end a
+                  |
+                  |type K = Map
+                  |""".stripMargin
+    runTestAssert[Source](code, assertLayout = None)(
+      Source(
+        List(
+          Defn.Object(Nil, Term.Name("a"), Template(Nil, Nil, Self(Name(""), None), Nil)),
+          Term.EndMarker(Term.Name("a")),
```

Comment:
Should the end marker by a separate entity? Shouldn't we maybe have `TemplateWithEnd` and `BlockWithEnd` ?

Or alternatively, we could just ignore it in the parser? Especially I imagine when doing some scalafix rules or scalafmt formatting we will not want to have to search for the end token.

Reply:
As discussed offline for now let's keep it separated and get back to it when we have significant indentation implemented and if `TemplateWithEnd` / `BlockWithEnd` are a way to go.